### ISSUE TITLE
cf push no longer errors when invalid path is supplied (with -p) and --no-start is specified

### ISSUE THREAD

**Issue by aauthor** [cloudfoundry/cli#733]
In previous versions, the CLI `FAILED` with exit code 1 when trying to push an app with `-p` with an invalid path (and `--no-start`) specified; however, there appears to be a change with the most recent version.  Now the CLI goes clear through binding a route, does not raise an error and exits with return code 0; however, if you try to start the app you'll get a cloud controller error `Server error, status code: 400, error code: 150001, message: The app package is invalid: bits have not been uploaded`.  See below for old behavior vs. new behavior.  Was this change intentional or is it a bug?

Old Behavior (cf version 6.14.1+dc6adf6-2015-12-22)
```
$ cf push fake-app -p fake/dir --no-start
Creating app fake-app in org appdirect-integration-suite / space development-15 as cf-pws-eng+appdirect-tests@pivotal.io...
OK

Creating route fake-app.cfapps.io...
OK

Binding fake-app.cfapps.io to fake-app...
OK

FAILED
Error processing app files in 'fake/dir': lstat /Users/pivotal/workspace/bam-ci/fake/dir: no such file or directory
```

New Behavior (cf version 6.15.0+fa1bfe2-2016-01-13)
```
$ cf push fake-app -p fake/dir --no-start
Creating app fake-app in org appdirect-integration-suite / space development-15 as cf-pws-eng+appdirect-tests@pivotal.io...
OK

Creating route fake-app.cfapps.io...
OK

Binding fake-app.cfapps.io to fake-app...
OK
```

Thanks!
Tom Dunlap && Tom Viehman

**Comment by dkoper** [comment]
Thanks Toms, we'll investigate.

**Comment by dkoper** [comment]
This has been implemented in cf CLI v6.16.0.
https://github.com/cloudfoundry/cli#downloads

Regards,
Dies Koper
CF CLI PM
